src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidDeadConditionalCode_NullAnalysis.cs

```diff
@@ -7474,5 +7474,51 @@ public static List<T> Test1<T>(IEnumerable<T> source)
                 LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp8,
             }.RunAsync();
         }
+
+        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.PointsToAnalysis)]
+        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.NullAnalysis)]
+        [Fact, WorkItem(6048, "https://github.com/dotnet/roslyn-analyzers/issues/6048")]
+        public async Task RethrowInCatchClause_NoDiagnosticsAsync()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+using System;
+
+class Test
+{
+    public static void M()
+    {
+        try
+        {
+            throw new Random().NextDouble() > 0.5 ? new Exception() : new ArgumentException();
+        }
+        catch (Exception e)
+        {
+            if (e is ArgumentException ae) // can be both true or false depending on the Exception thrown from the try-block, yet is flagged with CA1508 
+                Console.Out.WriteLine(""ArgumentException"");
+
+            throw;
+        }
+    }
+
+    public static void M2()
+    {
+        try
+        {
+            throw new Exception();
+        }
+        catch (Exception e)
+        {
+            if (e is ArgumentException ae)
```

Comment:
Maybe add a test with `catch (ArgumentException)` and asserting that the condition is flagged as "always true"?

Reply:
Sure, let me add it in the next PR.

Reply:
@Youssef1313 It seems this is a false negative right now. I'll file a tracking issue for it

```csharp
using System;

class Test
{
    public static void M()
    {
        try
        {
            throw new Random().NextDouble() > 0.5 ? new Exception() : new ArgumentException();
        }
        catch (ArgumentException e)
        {
            if (e is ArgumentException ae)
                Console.Out.WriteLine(""ArgumentException"");

            throw;
        }
    }
}
```